### PR TITLE
feat: display ERRORS first on validator documentation

### LINKED ISSUE #1605: Display ERRORS first on validator documentation
On the validator documentation, the order is currently: 
- WARNINGS
- ERRORS
- INFO

Screenshot 2023-11-20 at 3 54 41 PM

We should modify it so that ERRORS appear first.

### PR DESCRIPTION
**Pull request by cka-y**
**Summary:**
Closes #1605 

**Expected behavior:** 
`ERROR` notices are displayed first, followed by `WARNING` and then `INFO` in both the documentation and validation report.

Screenshot 2024-02-09 at 2 33 56 PM

Screenshot 2024-02-27 at 1 34 49 PM

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Add or update any needed [documentation](https://github.com/MobilityData/gtfs-validator/tree/master/docs) to the repo 
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [x] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

**Review by qcdyx — APPROVED** [review]
LGTM

**Comment by cka-y** [comment]
> @cka-y It looks like this original order is also affecting the actual HTML report (it's warning, info, error right now). I assume this would be a different issue - is that correct?

I included it as part of this PR

**Review by emmambd — APPROVED** [review]
LGTM after testing order on a few feeds

File: model/src/main/java/org/mobilitydata/gtfsvalidator/notice/SeverityLevel.java [hunk 1]
```diff
@@ -38,5 +38,18 @@ public enum SeverityLevel {
    * (e.g., using the language "must"). The validator uses RFC2119
    * (https://tools.ietf.org/html/rfc2119) to interpret the language in the GTFS spec.
    */
-  ERROR
+  ERROR;
+
+  public static int getOrder(SeverityLevel severityLevel) {
```
Comment (jcpitre) [inline model/src/main/java/org/mobilitydata/gtfsvalidator/notice/SeverityLevel.java:43]: Is it not more typical to put values in the enum elements themselves and just return that value?

**Review by jcpitre — APPROVED** [review]
I asked a question on a minor point, but could be merged as-is.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/report/model/ReportSummary.java [hunk 2]
```diff
@@ -45,7 +42,7 @@ public ReportSummary(NoticeContainer container, VersionInfo versionInfo) {
             .collect(
                 Collectors.groupingBy(
                     NoticeView::getSeverityLevel,
-                    LinkedHashMap::new,
+                    () -> new TreeMap<>(Comparator.comparingInt(SeverityLevel::getOrder)),
```
Comment (davidgamez) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/report/model/ReportSummary.java:45]: My understanding is the Enums already implements comparable having the natural order the order of the constant declaration. As the declaration order is INFO, WARNING and ERROR, we can use the reserve order to get the desired sorting, can you try here?
 ```suggestion
                    () -> new TreeMap<>(Comparator.reserveOrder()),
```

Comment (cka-y) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/report/model/ReportSummary.java]: it works ✅

File: model/src/main/java/org/mobilitydata/gtfsvalidator/notice/SeverityLevel.java (hunk 1 above)
Comment (cka-y) [reply model/src/main/java/org/mobilitydata/gtfsvalidator/notice/SeverityLevel.java]: Yes. As David pointed out it was much easier to just use the enum's reverse order.

**Review by davidgamez — APPROVED** [review]
LGTM!

**Pull request merged**